jwst/pipeline/calwebb_spec3.py

```diff
@@ -138,9 +138,19 @@ def process(self, input):
         for member in product['members']:
             members_by_type[member['exptype'].lower()].append(member['expname'])
 
-        if is_moving_target(input_models):
+        if is_moving_target(input_models[0]):
             self.log.info("Assigning WCS to a Moving Target exposure.")
-            input_models = self.assign_mtwcs(input_models)
+
+            # for compatibility with calwebb_image3, need to convert to ModelLibrary then back here
+            # keep asn metadata from input container - only metadata of individual models is modified
+            # by the assign_mtwcs step
+            library = ModelLibrary(input_models, on_disk=False)
+            library = self.assign_mtwcs(input_models)
+            with library:
+                for i, model in enumerate(library):
+                    input_models[i] = model.copy()
+                    library.shelve(model, modify=False)
+            del library
```

Comment:
```suggestion
            # assign_mtwcs modifies input_models in-place
            self.assign_mtwcs(input_models)
```
I think this should work. `assign_mtwcs` will convert the `input_models` `ModelContainer` to a `ModelLibrary` (but referencing the same models). As it modifies the models in-place the models in `input_models` are also updated.